src/test/java/net/snowflake/client/jdbc/ResultSetIT.java

```diff
@@ -1056,4 +1056,19 @@ public void testNextNegative() throws SQLException {
       }
     }
   }
+
+  /** SNOW-1416051; Added in > 3.16.0 */
+  @Test
+  public void shouldSerializeArrayAsObject() throws SQLException {
+    try (Connection connection = init();
+        Statement statement = connection.createStatement(); ) {
+      statement.execute("create or replace table test(ID NUMBER(38,0), A_ARRAY ARRAY)");
+      statement.execute("insert into test (id, a_array) select 1, ARRAY_CONSTRUCT(1,2,3)");
```

Comment:
We can even test it without inserting, doing select in place?

Reply:
Yes, we can - I wanted to follow the steps from the issue but indeed not all steps are necessary